include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -396,7 +396,18 @@ class CommonOperatorLoader : public ProtobufLoader {
                       "Can't register more than outputs in the operation.");
     numOutputs = (numOutputs < 0) ? op.output_size() : numOutputs;
     for (int i = 0; i < numOutputs; i++) {
-      nodeValueByName_[op.output(i)] = NodeValue(node, i);
+      // If output name is specified to run in FP16 precision and output type
+      // is fp16 add convert to fp32 node and update nodeValueByName_ map,
+      // so that next operator in the model receives appropriate nodeValue.
+      NodeValue output = node->getNthResult(i);
+      if (isOpRequestedInFP16Precision(op.output(i)) &&
+          output.getElementType() == ElemKind::Float16Ty) {
+        std::string opName = op.output(i) + "_convToFP32";
+        auto *CT = G_->createConvertTo(opName, output, ElemKind::FloatTy);
+        nodeValueByName_[op.output(i)] = CT->getResult();
+      } else {
+        nodeValueByName_[op.output(i)] = NodeValue(node, i);
```

Comment:
This logic is used in many places -- can we turn it into its own function that's called from everywhere you copied it?

Reply:
Done